src/pip/_internal/models/index.py

```diff
@@ -2,14 +2,19 @@
 
 
 class Index(object):
-    def __init__(self, url):
+    def __init__(self, url, link_source=''):
```

Comment:
You can drop the default on this.

Making it mandatory to pass doesn't affect anything and prevents from accidentally missing it out. :)